st/clitests/awss3api_spec.py

```diff
@@ -1841,6 +1853,335 @@ def get_md5_sum(fname):
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+check = all(item in common_prefixes for item in expected_common_prefixes)
+if check is False:
+    assert False, "Failed to match expected common prefix in the list"
+# Verify that NextToken is in the response and it is not empty
+assert (("next_token" in list_versions_response.keys()) and (len(list_versions_response["next_token"].strip()) > 0)), "NextToken is either not present or empty"
+next_token = list_versions_response["next_token"]
+
+# Step 9.2: command:= aws s3api list-object-versions --bucket <bucket> --max-items 2 --prefix "boo" --delimiter "/" --starting-token <next_token>
+# Expected part2 output:
+#  boo+
+list_versions_options = {"prefix":prefix, "delimiter":delimiter, "max-items":2, "starting-token":next_token}
+result = AwsTest(('Aws list object versions using prefix \'%s\', delimiter \'%s\' and starting-token %s') % (prefix, delimiter, next_token))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test()\
+    .command_is_successful()
+# Process result set
+list_versions_response = get_aws_cli_object(result.status.stdout)
+if list_versions_response is None:
+    assert False, "Failed to list object versions"
+# Get keys from the response
+object_versions = list_versions_response["versions"]
+print(f"Response: {list_versions_response} Versions received: {object_versions} Common prefixes received: {common_prefixes}")
+check = all(item in object_versions for item in expected_versions_list_part2)
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+
+# Step 10: Test negative max-keys value
+# Command:= aws s3api list-object-versions --bucket <bucket> --max-keys -10
+# Expected output: Invalid parameter error is returned
+list_versions_options = {"page-size":-10}
+result = AwsTest(('Aws list object versions using max-keys %s') % (-10))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test(negative_case=True)\
+    .command_should_fail()\
+    .command_error_should_have("InvalidArgument")\
+
+# Step 11: Test invalid encoding-type value
+# Command:= aws s3api list-object-versions --bucket <bucket> --encoding-type abcd
+# Expected output: Invalid parameter error is returned
+encoding = "abcd"
+list_versions_options = {"encoding":encoding}
+result = AwsTest(('Aws list object versions using encoding-type %s') % (encoding))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test(negative_case=True)\
+    .command_should_fail()\
+    .command_error_should_have("InvalidArgument")\
+
+# Step 12: Delete all created objects in bucket 'versionlistbucket'
+AwsTest('Aws delete objects')\
+    .delete_multiple_objects("versionlistbucket", object_list_file)\
+    .execute_test()\
+    .command_is_successful()\
+    .command_response_should_be_empty()
+
+delete_object_list_file(object_list_file)
```

Comment:
What's the point of calling this function if you're deleting the directory with `rm -rf` anyways?

Reply:
It is a different file out of that directory.